app/Http/Controllers/NewTeamController.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Enums\RoleName;
+use App\Http\Requests\Team\NewTeamRequest;
+use App\Models\Role;
+use App\Models\Team;
+use Illuminate\Http\RedirectResponse;
+use Illuminate\Support\Facades\Auth;
+use Inertia\Inertia;
+use Inertia\Response;
+
+class NewTeamController extends Controller
+{
+    public function index(): Response
+    {
+        return Inertia::render('Auth/NewTeam');
+    }
+
+    public function create(NewTeamRequest $request): RedirectResponse
+    {
+        $validated = $request->validated();
+
+        // check if there is a logo/image attached to the form
+        // (logo is not required, it can be set afterwards on the team profile page)
+        if ($request->logo) {
+
+            $path = public_path('logos/');
+
+            // if directory logos does not exists, we create one
+            !is_dir($path) && mkdir($path, 0777, true);
+
+            //set the name of the image/logo file
+            $logoName = time() . '.' . $request->logo->extension();
+
+            //move the image/logo to public folder->logos/
+            $request->logo->move($path, $logoName);
+
+        }
+
+        $team = new Team();
+        $team->name = $request->name;
+
+        if ($request->logo && $logoName) {
```

Comment:
You don't need to check for the logo name, if there's no file uploaded, it can be null by default.

Reply:
In order for the file to be reused in multiple dimensions (small, big, thumbnail etc) for multiple places all over the application, it may be good to implement a `webp` and JPEG file conversion queued job that will generate multiple files.

I don't know the dimensions off top of my head but if you could research what are the most used dimensions and file types for a web app, that would be awesome.

Reply:
I know that this was not mentioned in the ticket but you could do it in this ticket or we can do it later, your choice.